feat: dashboard lumo styles

tomivirkki opened the pull request:
## Description

First iteration of dashboard Lumo theme.
Visual tests are to be added separately once the component design is finalized.

Part of https://github.com/orgs/vaadin/projects/70/views/1?pane=issue&itemId=75293217

## Type of change

Feature

tomivirkki commented on packages/dashboard/theme/lumo/vaadin-dashboard-widget-styles.js:
```diff
@@ -0,0 +1,230 @@
+  /* Title slot styling */
+  slot[name='title']::slotted(*) {
+    flex: 1;
+    font-size: var(--lumo-font-size-m) !important;
+    font-weight: 500;
+    color: var(--lumo-header-text-color) !important;
```
Using `!important` here is a temporary workaround. We're considering moving the title element inside the component's shadow root instead.

ugur-vaadin commented:
There seems to be an issue with dragging an unfocused item in Firefox with this change. I cannot reproduce it using the feature branch.

tomivirkki commented:
FIxed, also fixed the following as discussed internally
- moved the title element inside the shadow root of widget/section
- used `<vaadin-dashboard-button>` instead of `<vaadin-button>`

ugur-vaadin reviewed (approved):
We should also remove the mention of title slots from the JSDocs of widget and section. Otherwise LGTM. One more review from someone would be good to have before we merge.

tomivirkki commented on packages/dashboard/src/vaadin-dashboard-item-mixin.js:
```diff
@@ -182,38 +206,58 @@ export const DashboardItemMixin = (superClass) =>
           id="resize-grow-width"
-        ></button>
-        <button
+          part="resize-grow-width-button"
```
The terminology is the same that's used for the accessible names/i18n, originating from [here](https://docs.google.com/spreadsheets/d/1cVXgRLIPqF0kFWKGTAmSITFRcizajP_pc9d5jEl9OvQ/edit?gid=523479405#gid=523479405). Let's discuss changing it separately.
